src/webui/api/appcontroller.cpp

```diff
@@ -275,8 +275,7 @@ void AppController::setPreferencesAction()
         }
 
         // Update deleted folders
-        for (auto i = oldScanDirs.cbegin(); i != oldScanDirs.cend(); ++i) {
-            QString folder = i.key();
+        for (const auto &folder : copyAsConst(oldScanDirs.keys())) {
```

Comment:
Why did you change this one? I don't see container changed inside loop.

Reply:
Since the danger of invalidating iterator comes from remove usages, I prefer reverting this one too, but I don't insist.

Reply:
Maybe we should give up such over-optimizations, and never return constant references on internal containers, but only its shared copies? This prevents potentially dangerous situations, as in the case above.

Reply:
>This prevents potentially dangerous situations, as in the case above.

+1

Reply:
Just to be clear. This loop seems to be ok before the change, right? I can't find an instance where `oldScanDirs` gets modified inside it. Am I right?

Reply:
>Am I right?

Yes.